src/lib/enhancers/createShowMore.ts

```diff
@@ -0,0 +1,69 @@
+interface ShowMoreState {
+  isShowingMore: boolean;
+  limit: number | null;
+  showMoreLimit: number | null;
+}
+
+interface ShowMore {
+  /**
+   * Toggles the widget to show more or less values.
+   */
+  toggleShowMore(): void;
+  /**
+   * Sets the listener function to call whenever `toggleShowMore` is called.
+   * It returns a function to unsubscribe the listener.
+   */
+  setToggleShowMore(listener: Listener): () => void;
+  /**
+   * Returns whether the widget is currently showing more values.
+   */
+  isShowingMore(): boolean;
+  /**
+   * Returns the current facets limit (`limit` or `showMoreLimit`)
+   */
+  getCurrentLimit(): number | null;
+}
+
+type CreateShowMore = (initialState?: Partial<ShowMoreState>) => ShowMore;
+type Listener = () => void;
+
+export const createShowMore: CreateShowMore = initialState => {
+  let onToggle: Listener | null = null;
+
+  const state: ShowMoreState = {
+    isShowingMore: false,
+    limit: null,
+    showMoreLimit: null,
+    ...initialState,
+  };
+
+  function toggleShowMore() {
+    state.isShowingMore = !state.isShowingMore;
+
+    if (onToggle) {
+      onToggle();
```

Comment:
where is onToggle used?

Reply:
That's the subscriber of the enhancer. It's the render method bound to its render params in the widgets.